spec/requests/presence_controller_spec.rb

```diff
@@ -122,58 +122,69 @@
+        }
+      )
 
       ch1.present(user_id: user.id, client_id: SecureRandom.hex)
       ch1.present(user_id: user2.id, client_id: SecureRandom.hex)
       ch1.present(user_id: user3.id, client_id: SecureRandom.hex)
 
-      get "/presence/get", params: { channel: ch1.name }
-      body = response.parsed_body
-      expect(body["users"].map { |u| u["id"] }).to contain_exactly(user.id, user2.id, user3.id)
-      expect(body["users"][0].keys).to contain_exactly("avatar_template", "id", "name", "username")
-      expect(body["count"]).to eq(3)
-      expect(body["last_message_id"]).to eq(MessageBus.last_id(ch1.message_bus_channel_name))
+      get "/presence/get", params: { channels: [ch1.name] }
+      expect(response.status).to eq(200)
+      state = response.parsed_body[ch1.name]
+      expect(state["users"].map { |u| u["id"] }).to contain_exactly(user.id, user2.id, user3.id)
+      expect(state["users"][0].keys).to contain_exactly("avatar_template", "id", "name", "username")
+      expect(state["count"]).to eq(3)
+      expect(state["last_message_id"]).to eq(MessageBus.last_id(ch1.message_bus_channel_name))
     end
 
     it "respects the existence/security of the channel" do
       sign_in user
 
-      get "/presence/get", params: { channel: ch1.name }
+      get "/presence/get", params: { channels: [ch1.name] }
       expect(response.status).to eq(200)
+      expect(response.parsed_body[ch1.name]).not_to eq(nil)
 
-      get "/presence/get", params: { channel: secure_user_channel.name }
-      expect(response.status).to eq(404)
+      get "/presence/get", params: { channels: [secure_user_channel.name] }
+      expect(response.status).to eq(200)
+      expect(response.parsed_body).to eq({ secure_user_channel.name => nil })
 
-      get "/presence/get", params: { channel: secure_group_channel.name }
-      expect(response.status).to eq(404)
+      get "/presence/get", params: { channels: [secure_group_channel.name] }
+      expect(response.status).to eq(200)
+      expect(response.parsed_body).to eq({ secure_group_channel.name => nil })
 
-      get "/presence/get", params: { channel: allowed_user_channel.name }
+      get "/presence/get", params: { channels: [allowed_user_channel.name] }
       expect(response.status).to eq(200)
+      expect(response.parsed_body[allowed_user_channel.name]&.keys).to include("users")
 
-      get "/presence/get", params: { channel: allowed_group_channel.name }
+      get "/presence/get", params: { channels: [allowed_group_channel.name] }
       expect(response.status).to eq(200)
+      expect(response.parsed_body[allowed_group_channel.name]&.keys).to include("users")
 
-      get "/presence/get", params: { channel: "/test/nonexistent" }
-      expect(response.status).to eq(404)
+      get "/presence/get", params: { channels: ["/test/nonexistent"] }
+      expect(response.status).to eq(200)
+      expect(response.parsed_body).to eq({ "/test/nonexistent" => nil })
     end
```

Comment:
Minor note about the assertions above in this tests. I feel like we can merge all of this into a single request since we now accept multiple channel names. The response also makes it easy for us to test each channel individually so we don't really need multiple requests here.

Reply:
Done in a099e38b0c308f203fab0616dde66c3ebf727785